File: Homeworks/C# Part 2/01.Arrays/11.BinarySearch/BinarySearch.cs
using System;

class BinarySearch
{
    static int BiSearch(int[] array, int searchedNumber, int start, int end)
    {
        for (int i = start; i <= end; i++)
        {
            if (array[i] == searchedNumber)
            {
                return i;                
            }            
        }
        return -1;
    }

    static void Main()
    {
        int numberN = int.Parse(Console.ReadLine());
        int[] numbers = new int[numberN];
        for (int i = 0; i < numbers.Length; i++)
        {
            numbers[i] = int.Parse(Console.ReadLine());
        }
        int numberX = int.Parse(Console.ReadLine());
        int firstIndex = 0;
        int lastIndex = numbers.Length - 1;
        int resultIndex = 0;
        for (int i = firstIndex; i < lastIndex; i++)
        {
            int middle = lastIndex / 2;
            if (numbers[middle] == numberX)
            {
                resultIndex = middle;
                break;
            }
            if (numbers[middle] < numberX)
            {
                firstIndex = middle;
                resultIndex = BiSearch(numbers, numberX, firstIndex, lastIndex);
            }
            if (numbers[middle] > numberX)
            {
                lastIndex = middle;
                resultIndex = BiSearch(numbers, numberX, firstIndex, lastIndex);
            }
        }
        if (resultIndex >= 0 && resultIndex < numbers.Length)
        {
            Console.WriteLine(resultIndex);
        }
        else
        {
            Console.WriteLine("-1");
        }
    }
}

File: Homeworks/JS Fundamentals/07. Arrays/07. Binary search/binary-search.js
function solve(args) {
    function biSearch(array, searchedNumber, start, end) {
        for (var i = start; i <= end; i += 1) {
            if (+array[i] === searchedNumber) {
                return i;
            }
        }
        return -1;
    }
    var data = args[0].split('\n'),
        numberN = +data.shift(),
        numberX = +data.pop();
    var firstIndex = 0;
    var lastIndex = data.length - 1;
    var resultIndex = 0;
    var middle = 0;
    for (var j = firstIndex; j < lastIndex; j += 1) {
        middle = (lastIndex / 2) | 0;
        if (+data[middle] === numberX) {
            resultIndex = middle;
            break;
        }
        if (+data[middle] < numberX) {
            firstIndex = middle;
            resultIndex = biSearch(data, numberX, firstIndex, lastIndex);
        }
        if (+data[middle] > numberX) {
            lastIndex = middle;
            resultIndex = biSearch(data, numberX, firstIndex, lastIndex);
        }
    }
    if (resultIndex >= 0 && resultIndex < data.length) {
        console.log(resultIndex);
    } else {
        console.log('-1');
    }
}